rfc_bibtex/rfc_bibtex.py

```diff
@@ -35,8 +36,32 @@ def __init__(self, id_names=[], in_file_name=None, out_file_name=None):
         if in_file_name is not None:
             self._id_names += self._read_ids_from_file(in_file_name)
 
+    @staticmethod
+    def _rfc_key_function(name):
+        """Turn RFC32 into RFC00032 for sorting"""
+        try:
+            if name.startswith('RFC'):
+                return "RFC"+format( int(name[3:]), "05d")
+        except (ValueError,TypeError) as e:
+            pass
+        return name
+
+    latex_citation_re = re.compile(r"^\\citation\{((rfc.*)|(draft-.*))\}",re.I)
     def _read_ids_from_file(self, file_name):
+        """
+        Read identifiers from a text file. 
+        If the text file is a LaTeX .aux file, parse the \citation{} command.
+        If the text file is a LaTeX .tex file, parse the corresponding .aux file
+        """
+        if file_name.endswith(".tex"):
+            file_name_aux = file_name[0:-4] + ".aux"
+            if not os.path.exists(file_name_aux):
```

Comment:
so you want file_name to be a Path object? How does one remove the extension and add another one? Or do you want me to check for it existing with pathlib?

Reply:
> so you want file_name to be a Path object?

Yes, that's correct. Using `pathlib` will make the code cleaner and more readable.

> How does one remove the extension and add another one? Or do you want me to check for it existing with pathlib?

```
if file_name.stem == '.tex':
    file_name_aux = file_name.parent / (file_name.stem + '.aux')
```

P.S. feel free to object to whatever I say and express your opinion. This is not a dictatorship :smile:

Reply:
> Also, do you want to use PathLib for opening files and reading lines as well? It really seems like overkill.

You mean at  `with open(file_name, 'r') as in_file`? Since you already have `file_name` as `Path`, you can just do `with file_name.open() as in_file:`. Why would that be an overkill?´

And since `pathlib` is being used, it's pretty easy and elegant to add checks if the file exists and that it's actually a file, and if no, give the user an nice error message. But don't worry too much about this one, it's not a critical addition :wink:

Reply:
I don't think that replacing short, concise, descriptive code with longer, less efficient, object-oriented code is normally a good thing to do.  I also think that pathlib's overriding of the `/` is a serious mistake. But I'll clean it up. Thanks for the code suggestion.

Reply:
Actually, it is a dictatorship, as you are the person who approves the pull requests.

Reply:
;-)

Reply:
It makes the code more readable and you can test both versions for speed, there won't be any significant difference :smile: 

I believe you haven't pushed the changed yet.